executor/executor.go

```diff
@@ -1797,8 +1801,8 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 			goCtx = pprof.WithLabels(goCtx, pprof.Labels("sql", util.QueryStrForLog(prepareStmt.NormalizedSQL)))
 			pprof.SetGoroutineLabels(goCtx)
 		}
-		if topsqlstate.TopSQLEnabled() && prepareStmt.SQLDigest != nil {
-			topsql.AttachSQLInfo(goCtx, prepareStmt.NormalizedSQL, prepareStmt.SQLDigest, "", nil, vars.InRestrictedSQL)
+		if prepareStmt.SQLDigest != nil {
+			AttachSQLInfoForTopSQL(goCtx, sc, prepareStmt.NormalizedSQL, prepareStmt.SQLDigest, vars.InRestrictedSQL)
```

Comment:
Why not use the returned `context.Context` here?

Reply:
Since `context.Context` won't be returned outside, And it's ok to not use the `context.Context`.